docs/source/datatypes/index.rst

```diff
@@ -25,7 +25,7 @@ the methods provided, how to access them, and so on.
 
 If you need to work with some specific type of data, first check the list of data types/plugins
 below, and if you don't find what you need, give a look to
-:ref:`how to write a new data plugin <working_data_creating_new_types>`.
+:ref:`how to write a new data plugin <how-to:data:plugin:create>`.
```

Comment:
It shouldn't, this content should be moved into the new scaffolding, but it is not in the scope of the issue addressed by this PR, so I left it and just made sure the link is already correct. Having said that, I realize now that there currently is not really an issue that addresses these sections. This content contains detailed information on explicit data types shipped with `aiida-core`, some of which will be moved out at some point. Not sure what to do with this documentation until that time

Reply:
Are those data types documented as part of the reference? In that case we can maybe just drop it. Alternatively we could place it to the end of HowTo::"How to work with data".

Reply:
the content of `dataypes/index.rst` is really minimal and could certainly be furnished by the reference section. This would then of course rely on comprehensive docstring on the data types itself in the code, which we would have to check. The other files in that folder go more in depth, and are mini tutorials on how to work with the more advanced data types. But this concerns mostly the data types that are materials science specific and so will be moved out at some point. I think Giovanni would prefer not to get rid of this documentation yet so maybe we discuss with him during the meeting what we do with them for the time being

Reply:
As discussed offline with @csadorf , at the end of the documentation restructuring we will look where to put the content of `datatypes/` and move it somewhere appropriate.